tests/framework/cli/test_catalog.py

```diff
@@ -360,3 +387,35 @@ def test_bad_env(self, fake_project_cli, fake_metadata):
 
         assert result.exit_code
         assert "Unable to instantiate Kedro session" in result.output
+
+
+@pytest.mark.usefixtures(
+    "chdir_to_dummy_project", "fake_load_context", "mock_pipelines"
+)
+def test_list_catalog_factories(
```

Comment:
Should we add some failure test cases for example, testing empty or invalid DataCatalog configuration?

Reply:
Good catch!

Following the test setup for `kedro catalog list` neither of these cases are tested for. Behaviour-wise an invalid catalog produces an error for both `kedro catalog list` and `kedro catalog factory list`. An empty catalog leaves all datasets in the pipeline to be described as a default dataset for `kedro catalog list`, and an empty set `[ ]` to be output by `kedro catalog factory list`. The same empty set is output for a valid catalog with no patterns.

 I think catching it and returning a "There are no dataset factory patterns in the catalog" message should suffice, and then adding a test case to ensure it's output where it should be.

Reply:
Added in [a8657db](https://github.com/kedro-org/kedro/pull/2796/commits/a8657db7e146de1e7b0a0667d13ba642d0ef30d8)